apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -166,6 +166,248 @@ handle_request('GetGenerationByHeight', Params, _Context) ->
         {ok, Hash} -> get_generation(Hash)
     end;
 
+handle_request('GetAccountByPubkey', Params, _Context) ->
+    case aec_base58c:safe_decode(account_pubkey, maps:get(pubkey, Params)) of
+        {ok, Pubkey} ->
+            case aehttp_logic:get_account(Pubkey) of
+                {ok, Account} ->
+                    {200, [],
+                     #{pubkey => aec_base58c:encode(account_pubkey, Pubkey),
+                       balance => aec_accounts:balance(Account),
+                       nonce => aec_accounts:nonce(Account)}};
+                {error, _} ->
+                    {404, [], #{reason => <<"Account not found">>}}
+            end;
+        {error, _} ->
+            {400, [], #{reason => <<"Invalid public key">>}}
+    end;
+
+handle_request('GetPendingAccountTransactionsByPubkey', Params, _Context) ->
+    case aec_base58c:safe_decode(account_pubkey, maps:get(pubkey, Params)) of
+        {ok, Pubkey} ->
+            case aec_chain:get_account(Pubkey) of
+                {value, _} ->
+                    {ok, Txs0} = aec_tx_pool:peek(infinity, Pubkey),
+                    Txs = [aetx_sign:serialize_for_client_pending(json, T) || T <- Txs0],
+                    JsonTxs = #{data_schema => <<"JSONTx">>,
+                                transactions => Txs},
+                    {200, [], JsonTxs};
+                _ ->
+                    {404, [], #{reason => <<"Account not found">>}}
+            end;
+        {error, _} ->
+            {400, [], #{reason => <<"Invalid public key">>}}
+    end;
+
+handle_request('GetTransactionByHash', Params, _Config) ->
+    case aec_base58c:safe_decode(tx_hash, maps:get(hash, Params)) of
+        {ok, Hash} ->
+            case aec_chain:find_tx_with_location(Hash) of
+                none ->
+                    {404, [], #{<<"reason">> => <<"Transaction not found">>}};
+                {mempool, Tx} ->
+                    JSONTx = aetx_sign:serialize_for_client_pending(json, Tx),
+                    {200, [], JSONTx};
+                {BlockHash, Tx} ->
+                    {ok, Header} = aec_chain:get_header(BlockHash),
+                    JSONTx = aetx_sign:serialize_for_client(json, Header, Tx),
+                    {200, [], JSONTx}
+            end;
+        {error, _} ->
+            {400, [], #{reason => <<"Invalid hash">>}}
+    end;
+
+handle_request('GetTransactionInfoByHash', Params, _Config) ->
+    ParseFuns = [read_required_params([hash]),
+                 base58_decode([{hash, tx_hash, tx_hash}]),
+                 get_transaction(tx_hash, tx),
+                 get_contract_call_object_from_tx(tx, contract_call),
+                 ok_response(
+                    fun(#{contract_call := Call}) ->
+                            aect_call:serialize_for_client(Call)
+                    end)
+                ],
+    process_request(ParseFuns, Params);
+
+handle_request('PostTransaction', #{'Tx' := Tx}, _Context) ->
```

Comment:
This seems the same as PostTx. Not refactoring because PostTx planned for removal?

Reply:
That's correct, we're going to remove it in following PRs.